platform/src/main/java/org/stellar/anchor/platform/service/PaymentOperationToEventListener.java

```diff
@@ -24,15 +26,29 @@ public class PaymentOperationToEventListener implements PaymentListener {
 
   @Override
   public void onReceived(PaymentOperationResponse payment) {
+    if (!payment.getTransaction().isPresent()) {
+      return;
+    }
+
+    MemoHash memoHash = (MemoHash) payment.getTransaction().get().getMemo();
+    String hash = new String(Base64.encodeBase64(memoHash.getBytes()));
+
     // Find the matching transaction
-    Sep31Transaction txn = transactionStore.findByStellarAccountId(payment.getTo());
+    Sep31Transaction txn = null;
+    try {
+      txn = transactionStore.findByStellarMemo(hash);
+    } catch (SepException e) {
+      Log.info(String.format("error finding transaction that matches the memo (%s).", hash));
+    }
+
     if (txn == null) {
-      Log.error(String.format("no transaction(stellarAccountId=%s) is found.", payment.getTo()));
+      Log.info(String.format("no transaction(stellarAccountId=%s) is found.", payment.getTo()));
       return;
     }
 
     if (!(payment.getAsset() instanceof AssetTypeCreditAlphaNum)) {
-      // Asset does not match
+      // Asset does not match. Ignore.
+      Log.info(String.format("unexpected payment type %s", payment.getAsset().getClass()));
```

Comment:
‼️   Seems like you're validating the stellar payment `memo` and `asset code`, but you're not validating the amount nor the asset issuer. This way, someone could send one stroop and e approved, which is pretty bad.

I think we should make sure the amount received is `>=` the amount expected.

Reply:
I'll address this one in the PR I've been working on, since this is not an issue you introduced but something that was already in the code.

Reply:
Yes, we should validate the asset issuer and the amount.